.travis.yml

```diff
@@ -1,2 +1,10 @@
 language: ruby
+sudo: required
+services:
+  - rabbitmq
+before_script:
```

Comment:
@JaKXz I don't understand the rationale behind this. Following is your original comment:

> This is a good change! However, I think the expensive end to end tests should happen after the less expensive unit tests, so I would use after_script or even after_success.

Because of [the way Travis decides if a build was a failure](https://docs.travis-ci.com/user/customizing-the-build/#Breaking-the-Build), if we run the end to end tests under `after_script` or `after_success` and these tests fail, the overall build will still pass if unit tests passed in `script` section.

This might end up giving us false positives. We can of course manually check the output of the tests run on Travis but the status of the build will be green in that case and a PR, if it causes the end to end tests to fail, will be okay to merge.

The end to end tests are expensive to run, yes. But if these are not going to affect the build status, I don't see the point of running them in the first place.